### ISSUE TITLE
addEventListener return type

### ISSUE THREAD

**Issue by cknitt** [reasonml-community/bs-react-native#408]
This is not consistent in the bindings.

I think addEventListener should always return an `EventSubscription.t`.

**Comment by sgny** [comment]
@MoOx I understand the intention behind the commit, and most `listener` subscription do not return `EventSubscription` objects, but what we had for `BackHandler` and `NetInfo` were correct, as they return `{"remove": unit => unit}`.

I think this commit should be reverted.

@cknitt What do you think?

**Comment by sgny** [comment]
Link to the [discussion](https://github.com/reasonml-community/bs-react-native/pull/403) before.

**Comment by MoOx** [comment]
I probably committed too quickly, that's won't be my first time haha.

Thanks, I read the discussions & think that `addEventListener` are going to be replaced by `addListener`.
I assumed this by reading this https://github.com/facebook/react-native/blob/d9930897a8ab80fb49628efb2a08d1e0c4d4fee6/Libraries/AppState/AppState.js#L65.

So we can assume this is still correct & would avoid us to have to do this commit again when they change this?
Otherwise should I just revert my commit & close this issue again?

**Comment by sgny** [comment]
No worries, the move to `reason-react-native` (much nicer than `bs-react-native-next`) had the side effect of making file histories hard to navigate. I remember only because I spent quite a bit of time trying to make sense of these objects. I got nowhere, however, as nothing is actually returned in most cases. I had not tested on Android at the time, perhaps that might give different results.

I think we should either revert this commit and handle `AppState` with another commit (perhaps only bindings to `addListener` and `removeListener`) or add a getter for `NetInfo` and `BackHandler` as I suggested in that discussion.
